force-app/main/default/classes/NKS_ConversationEntryController.cls

```diff
@@ -0,0 +1,272 @@
+                message.CRM_From_NAV_Unit__c = navUser.Department;
+            }
+        } else {
+            message.CRM_From_Contact__c = session.EndUserContactId;
+        }
+
+        return message;
+    }
+
+    private static void insertMessagesAndUpdatedThreadCloseDate(List<Message__c> messages, String threadId) {
+        try {
+            insert messages;
+            updateThreadCloseDate(threadId, System.now());
+        } catch (Exception e) {
+            logger.error('Error saving messages: ' + e.getMessage(), null, CRM_ApplicationDomain.Domain.NKS);
+        }
+    }
+
+    private static void splitChatbotMessagesFromConversation(
+        List<ConversationEntry> entries,
+        List<ConversationEntry> chatbotMessages,
+        List<ConversationEntry> userMessages
+    ) {
+        Boolean automaticGreetingFound = false;
+        for (ConversationEntry entry : entries) {
+            if (entry.messageText == null) {
+                continue;
+            }
+            entry.messageText = entry.messageText.unescapeHtml4(); // Connect API returns HTML-encoded strings -> decode for further processing
+
+            if (isWelcomeMessage(entry)) {
+                automaticGreetingFound = true;
+                entry.sender.role = 'Agent'; // Welcome message is incorrectly set as User message
+                userMessages.add(entry); // Add welcome message to user messages
+                continue;
+            }
+
+            if (automaticGreetingFound) {
+                userMessages.add(entry);
+            } else {
+                chatbotMessages.add(entry);
+            }
+        }
+    }
+
+    // TODO: Only check sender.role == System to avoid this failing if we change the welcome message in the future?
+    private static Boolean isWelcomeMessage(ConversationEntry entry) {
+        return entry.messageText != null &&
+            'System'.equalsIgnoreCase(entry.sender.role) &&
+            (entry.messageText.containsIgnoreCase('Welcome to chat with NAV') ||
+            entry.messageText.containsIgnoreCase('Velkommen til chat med NAV'));
+    }
+
+    private static String formatChatbotMessagesAsString(List<ConversationEntry> entries) {
+        List<String> formattedMessages = new List<String>();
+
+        for (ConversationEntry entry : entries) {
+            String messageText = entry.messageText;
+            // Assuming user sent more messages after chat was sent to queue
+            if (!messageText.startsWith('Bruker:') && !messageText.startsWith('Frida:')) {
+                messageText = 'Bruker: ' + messageText;
+            }
+            formattedMessages.add(messageText);
+        }
+
+        return String.join(formattedMessages, '\n');
+    }
+
+    private static void storeChatbotConversationOnThread(String threadId, String chatbotConversation) {
+        update new Thread__c(Id = threadId, CRM_Additional_Information__c = chatbotConversation);
+    }
```

Comment:
Den her metoden burde ikke eksistere, det gjøres allerede update på thread seinar i updateThreadClosedate, så det blir dumt å kjøre en update av samme objekt to ganga i en transaction